### DISCUSSION TITLE
Should a semicolon be escaped?

### DISCUSSION THREAD

**Discussion by egh** [typst/typst#2960] [General]
I raised an issue with pandoc (here https://github.com/jgm/pandoc/issues/9252) where a semicolon `;` was not displayed. But pandoc seems to be outputting the correct thing, but it's not 100% clear to me.

What should this output?
```
#link("http://example.org")[foo]; bar
```

I would expect to see:
> foo; bar

where `foo` is a link. But the semicolon disappears. It works if escaped:
```
#link("http://example.org")[foo]\; bar
```

But I don't see where in the documentation it says `;` needs escaping, and it works when there is no link.

**Comment by Enivex** [comment]
It's because `#` brings you to code mode, where `;` indicates the end of an expression

**Reply by egh** [reply]
Thank you! So the correct thing is to terminate code mode with a `;` and then add another `;`?

**Reply by laurmaedje** [reply]
After code mode, yes, either that or escaping. Personally, I'd say escaping is a little more readable. Both are fine though.

**Reply by egh** [reply]
Thank you!
